### PR TITLE
Field labels: histogram option for numeric fields

### PR DESCRIPTION
**Pull request by gtk-grafana**
Adds a new option for numeric fields to allow users to toggle between histogram and time series visualization types.
Also adds grouping categories to the PanelMenu.
Saves last selection in local storage as default viz for numeric panels.

image
image

Without investigations:
Numeric panel:
image
Regular panel:
image

Fixes: https://github.com/grafana/explore-logs/issues/923

### CODE REVIEW

File: src/Components/Panels/PanelMenu.tsx [hunk 1]
```diff
@@ -19,16 +21,77 @@ import { logger } from '../../services/logger';
...
+      const gridItem = sceneGraph.getAncestor(sceneRef, SceneCSSGridItem);
+      const viz = sceneGraph.getAncestor(sceneRef, VizPanel).clone();
+      const $data = sceneGraph.getData(sceneRef).clone();
+      const menu = sceneRef.clone();
+      const headerActions = Array.isArray(viz.state.headerActions)
+        ? viz.state.headerActions.map((o) => o.clone())
+        : viz.state.headerActions;
+      let body;
+
+      if (sceneRef.state.panelType !== AvgFieldPanelType.histogram) {
```
Comment (matyax) [inline src/Components/Panels/PanelMenu.tsx:68]: Can we add event tracking for this?

File: src/services/store.ts [hunk 2]
```diff
@@ -231,3 +234,16 @@ export function getLogsVisualizationType(): LogsVisualizationType {
...
   localStorage.setItem(VISUALIZATION_TYPE_LOCALSTORAGE_KEY, type);
 }
+
+// Panel options
+const PANEL_OPTIONS_LOCALSTORAGE_KEY = `${pluginJson.id}.panel.option`;
+export interface PanelOptions {
+  panelType: AvgFieldPanelType;
+}
+export function getPanelOption<K extends keyof PanelOptions, V extends PanelOptions[K]>(option: K): V | null {
+  return localStorage.getItem(`${PANEL_OPTIONS_LOCALSTORAGE_KEY}.${option}`) as V | null;
```
Comment (matyax) [inline src/services/store.ts:244]: This could probably use integration with narrowing.ts

File: src/services/store.ts [hunk 3]
```diff
@@ -214,6 +201,22 @@ export function setDisplayedFields(sceneRef: SceneObject, fields: string[]) {
   localStorage.setItem(`${pluginJson.id}.${PREFIX}.logs.fields`, JSON.stringify(fields));
 }
 
+// Log panel options
```
Comment (matyax) [inline src/services/store.ts:204]: Were these 2 functions just moved?

**Review by matyax — APPROVED** [review]
Looks good and works as expected! Would appreciate if you add usage tracking and sync with narrowing.ts changes before

File: src/services/store.ts (hunk 3 above)
Comment (gtk-grafana) [reply src/services/store.ts:204]: Yeah, might be good to break out the store a bit in the future, I was just organizing a bit

File: src/services/store.ts (hunk 2 above)
Comment (gtk-grafana) [reply src/services/store.ts]: Good catch, I removed the assertion and updated the getPanelOption to require providing all possible values for the panel options, might not work for options with many/infinite possible values, but provides better safety for these enum options.

**Pull request merged**